diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -313,6 +314,14 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
     private fun ASTNode.getParentExpressions() =
             parents().takeWhile { it.elementType in chainExpressionTypes && it.elementType != LAMBDA_ARGUMENT }
 
+    private fun ASTNode.containsManyFunctions() =
```

Comment:
Initial attempt on this was so that chain of property accesses (without parenthesis) was also considered part of functional style. If I understand correctly, if you count only function calls, you will allow things like
```
list
    .filter {
    }
    .foo.bar.baz
    .forEach { }
```
I suggest instead to only filter out non-call parts of dot qualified expression at the beginning. Everything after the first call in the chain cannot be a fully qualified name.